openstudiocore/src/model/GeneratorMicroTurbine.hpp

```diff
@@ -0,0 +1,249 @@
+/**********************************************************************
+ *  Copyright (c) 2008-2016, Alliance for Sustainable Energy.
+ *  All rights reserved.
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ **********************************************************************/
+
+#ifndef MODEL_GENERATORMICROTURBINE_HPP
+#define MODEL_GENERATORMICROTURBINE_HPP
+
+#include <model/ModelAPI.hpp>
+#include "Generator.hpp"
+
+namespace openstudio {
+
+namespace model {
+
+class Curve;
+// For the optional Generator:MicroTurbine:HeatRecovery
+// It was broken out because that part needs to connect to a plant loop
+class StraightComponent;
+
+// TODO: add the tables class if they get added to OS later?
+//class DataTables // UniVariateTables and BiVariateTables
+
+//class Connection;
+
+namespace detail {
+
+  class GeneratorMicroTurbine_Impl;
+
+} // detail
+
+/** GeneratorMicroTurbine is a Generator that wraps the OpenStudio IDD object 'OS:Generator:MicroTurbine'. */
+class MODEL_API GeneratorMicroTurbine : public Generator {
+ public:
+  /** @name Constructors and Destructors */
+  //@{
+
+  explicit GeneratorMicroTurbine(const Model& model);
+  
+  // TODO: Could add a constructor with the curve but that'd be nuts, there are 13 of them
+
+  virtual ~GeneratorMicroTurbine() {}
+
+  //@}
+
+  static IddObjectType iddObjectType();
+
+  static std::vector<std::string> validFuelTypeValues();
+
+  /** @name Getters */
+  //@{
+
+  double referenceElectricalPowerOutput() const;
+
+  double minimumFullLoadElectricalPowerOutput() const;
+  bool isMinimumFullLoadElectricalPowerOutputDefaulted() const;
+
+  boost::optional<double> maximumFullLoadElectricalPowerOutput() const;
```

Comment:
I would make this return a double, if it is not set just return referenceElectricalPowerOutput like E+ does.  You would also add an isMaximumFullLoadElectricalPowerOutputDefaulted